src/crimson/osd/objclass.cc

```diff
@@ -123,6 +123,25 @@ int cls_cxx_stat2(cls_method_context_t hctx,
                   uint64_t *size,
                   ceph::real_time *mtime)
 {
+  OSDOp op{CEPH_OSD_OP_STAT};
+  if (const auto ret = execute_osd_op(hctx, op); ret < 0) {
+    return ret;
+  }
+  ceph::real_time ut;
+  uint64_t s;
```

Comment:
I think a better way would be to use assert on `size` and `mtime` at the first line of the function and then using them directly when calling decode. No need to use dummy variables. I just pushed this change, let me know what you think.

Reply:
we do have use cases where a null `size` or null `mtime` is passed down. please grep the source for the callers.

Reply:
Ok I agree that assert would not work here. However, I am not sure the suggested change achieves the purpose of the method: If `size` is not nullptr then the decoded value is supposed to be assigned to `*size`. What is the purpose of `out_size` in your suggested change?

Reply:
instead of writing down the whole solution, i wanted to put part of it so you could ponder on it, and create the missing puzzle by yourself.
```c++
  uint64_t dummy_size;
  ceph::real_time dummy_mtime;
  uint64_t& out_size = size ? *size : dummy_size;
  ceph::real_time& out_mtime = mtime ? *mtime : dummy_mtime;
  try {
    auto iter = op.outdata.cbegin();
    decode(out_size, iter);
    decode(out_mtime, iter);
    return 0;
  } catch (buffer::error& err) {
    return -EIO;
  }
```

Reply:
Pushed the changed. Thanks.